### PR TITLE
Fix: There's a lot of extra space above the report actions

### PR DESCRIPTION
**Pull request by rayane-d**
### Fixed Issues

$ https://github.com/Expensify/App/issues/35921
$ https://github.com/Expensify/App/issues/36166
PROPOSAL: N/A

### Tests

1. Open a new chat.
2. Ensure that the empty state background image is visible in the empty space above the first report action, and confirm it does not occupy the scrollable area.
3. Verify that the image overlaps the first report action by 60px.
4. Send messages.
5. Verify that when space is limited, the background image is truncated at the top, preventing scrolling to view the entire image.

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A
### QA Steps

Same as above tests.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/77965000/e961325a-a1e8-4825-a9f2-8af34bebc29d

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/77965000/d0b6f992-fe42-433a-a735-fe0a84ba5c2f

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/77965000/d3b57969-aa2e-45e8-a181-f939f2d8e940

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/77965000/021bb39d-2dc8-4ec4-9408-f1803cf1e2c1

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/77965000/cf64cade-7e2f-4b20-99e4-aecac487c1c5

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/77965000/aeabb9f0-8110-4581-9bc0-3f1ca69a5779

</details>

### CODE REVIEW

File: src/pages/home/report/ReportActionItem.tsx
Comment (cubuspl42) [inline src/pages/home/report/ReportActionItem.tsx:699]: Does a single prop-less `View` make any sense in React Native? Leave alone two.

**Review by jakub-trzebiatowski — CHANGES_REQUESTED** [review]

File: src/pages/home/report/ReportActionItemCreated.tsx
Comment (cubuspl42) [inline src/pages/home/report/ReportActionItemCreated.tsx:63]: How does removing the animated empty-state background relate to our task?

File: src/pages/home/report/ReportActionItem.tsx
Comment (cubuspl42) [reply src/pages/home/report/ReportActionItem.tsx]: @rayane-djouah Does a _single_ prop-less `View` make any difference? I could imagine that in _some_ cases it might make a difference on Web, but I would still test it.

Unless you know or strongly suspect that it **is** needed, than please explain it to me.

File: src/pages/home/report/ReportActionItemCreated.tsx
Comment (cubuspl42) [reply src/pages/home/report/ReportActionItemCreated.tsx:63]: Thank you

File: src/pages/home/report/ReportActionItem.tsx
Comment (cubuspl42) [inline src/pages/home/report/ReportActionItem.tsx:714]: Another one.

**Review by jakub-trzebiatowski — CHANGES_REQUESTED** [review]

File: src/pages/home/report/ReportActionItem.tsx
Comment (cubuspl42) [reply src/pages/home/report/ReportActionItem.tsx]: In most cases, a fragment `<>` will behave the same, while it is a simpler concept than a `View`.

Comment (cubuspl42) [reply src/pages/home/report/ReportActionItem.tsx]: (of course, when multiple children are involved)

**Comment by cubuspl42** [comment]
@yuwenmemon Can we trigger an ad-hoc build, so the design changes can be reviewed easily?

**Comment by shawnborton** [comment]
Hmm I think the background image should scroll with the content, so that at some point the background image just scrolls off the view entirely.

**Comment by shawnborton** [comment]
Hmm this does not work at all on mobile web.

**Comment by shawnborton** [comment]
> Would it be possible to show the remaining part of the placeholder image (instead of a white rectangle) when scrolling beyond the base scrolling range, i.e. when I "pull the spring"?

Yup, totally agree with that!

**Comment by shawnborton** [comment]
> Unfortunately, it's not possible to display the remaining part of the placeholder image when scrolling beyond the base scrolling range.

Did you try removing overflow:hidden from the wrapper that has the background image? 

Also another small thing, if you zoom in it looks like the background image is getting cut off at the bottom whereas it's supposed to look like a clean fade:
CleanShot 2024-04-01 at 07 27 44@2x

**Comment by shawnborton** [comment]
> Also another small thing, if you zoom in it looks like the background image is getting cut off at the bottom whereas it's supposed to look like a clean fade:

Although maybe that's just a poor screenshot quality, because when I test on Web it looks fine.

**Comment by shawnborton** [comment]
Hmm but can't we use some kind of absolute positioning to make sure that element doesn't occupy scrollable space and it just kinda "floats" above the top-most content?

**Comment by cubuspl42** [comment]
I'll try experimenting on this.

**Comment by cubuspl42** [comment]
> @[rayane-djouah](https://github.com/rayane-djouah) rayane-djouah requested a review from https://github.com/orgs/Expensify/teams/design

@rayane-djouah How does the current state of the implementation relate to [the demo I provided](https://github.com/Expensify/App/issues/36166#issuecomment-2036577011) on the issue thread?

Are we really ready for another round of design review? 🤔

**Comment by cubuspl42** [comment]
I got it! Let me know if you have any concerns or doubts. The demo I provided is minimalistic and makes use of layout callbacks. It seems that there's no 100% direct (style-only) way to achieve the desired effect. Incorporating the approach from the demo into Expensify is a separate problem, and I didn't mean to suggest it's trivial.

**Comment by cubuspl42** [comment]
Yeah, I was afraid there might be some gotchas. I'll give it a look.

**Comment by cubuspl42** [comment]
I feel that you suggested that our flat list being "inverted" is the issue, but the problem begins earlier. In general, it's not really possible to achieve the effect from my demo when using `FlatList` (or something based on it), or at least I don't know such a way.

To work it around, we _could_ implement something like this...

```tsx
type FlatListFactory<ItemT> = {
    /**
     * Render a real flat list, without a measurement support.
     */
    renderFlatList(props: BasicFlatListProps<ItemT>): React.ReactElement,
};

type BasicFlatListProps<ItemT> = Pick<FlatListProps<ItemT>, 'ListHeaderComponent' | 'data' | 'renderItem'>; // and more...

type FlatListContentSizeChangeEvent = {
    /**
     * The height of the flat list's content (excluding the header)
     */
    readonly contentHeight: number;
}

type MeasurableFlatListProps<ItemT> = BasicFlatListProps<ItemT> & {
    /**
     * Called when scrollable content view of the flat list changes. Invoked only if the list is in the "scrollable" mode.
     */
    onContentSizeChange: (event: FlatListContentSizeChangeEvent) => void;
}

type HybridFlatListProps<ItemT> = MeasurableFlatListProps<ItemT> & {
    /**
     * If the item count is below this threshold, the list will be in the "scrollable" mode and will render a
     * `ScrollableFlatList`. Otherwise, it will be in the "real" mode and will render a real flat list.
     */
    itemCountThreshold: number,

    /**
     * A factory that allows rendering a real flat list if in the "real" mode.
     */
    realFlatListFactory: FlatListFactory<ItemT>,
};

/**
 * A simple implementation of a subset of the `FlatList` API, backed by the `ScrollView`. Supports content measurement.
 */
function ScrollableFlatList<TItem>({data, renderItem, onContentSizeChange}: MeasurableFlatListProps<TItem>) {
    return (
        <ScrollView
            onContentSizeChange={(width, height) => {
                // ...
            }}
        >
            {/* implementation... */}
        </ScrollView>
    )
}

/**
 * `onContentSizeChange` is invoked only if the list in in the "scrollable" mode.
 */
function HybridFlatList<TItem>(props: HybridFlatListProps<TItem>) {
    const {data, itemCountThreshold, realFlatListFactory} = props;

    if (data.length < itemCountThreshold) { // "scrollable" mode
        return <ScrollableFlatList {...props} />;
    } else { // "real" mode
        return realFlatListFactory.renderFlatList(props);
    }
}
```

We could quite easily implement `FlatListFactory<ItemT>` for `InvertedFlatList`.

I believe we could **then** implement `EmptyStateFlatList` based on `HybridFlatList`, benefiting from its `onContentSizeChange` and `ListHeaderComponent`.

There are some gotchas:
- What subset of `FlatListProps` would we need in `BasicFlatListProps`? The less, the easier it will be to implement `ScrollableFlatList` and the lower is the risk of some kind of "jump" during switching between the "scrollable" and "real" mode.
- Could we actually get all the overflow effects we need from `ListHeaderComponent`? Do we need to change the API to include some of our assumptions about overflows?
- This is getting much more complex than what I initially thought

**Comment by cubuspl42** [comment]
Okey, an update from my side:

I spent quite some time implementing a sketch of a new solution, assuming that `FlatList` doesn't support `onContentSizeChange`. In fact, it **does** (as its props extend `ScrollViewProps`), and likely that's what you used in this solution:

> I attempted the current method with FlatList, and it worked

Could you comment on how `onContentSizeChange` works on `FlatList`, according to your observations? Does it give strong guarantees (i.e. actually calculates the height of all data items), or does it handle it in some best-effort way, for example reporting only the actually rendered items? Whichever it is, are confident that it can work well for our needs in all cases?

> but the issue arises with InvertedFlatList.

I'm done for today, could you try hacking on the demo, reverting the math vertically where required? Or is it more complex than that?

**Comment by cubuspl42** [comment]
> @cubuspl42 I don't see the need for layout callbacks like `onContentSizeChange`. I removed these callbacks from your demo, and it still works as expected. I also tried it with `FlatList` without the callbacks, and it worked.

I'm very confused; those callbacks did something! 🤔 Please share a fork of my demo!

**Comment by cubuspl42** [comment]
Awesome! I'm exploring your implementations.

**Comment by cubuspl42** [comment]
> "Rubber banding" in the context of web design refers to a visual effect often used on mobile interfaces. When you try to scroll beyond the edge of a page or a list, the content will move as if it's being pulled by a rubber band, but it will bounce back into place when you let go. This gives the user a clear indication that they've reached the end of the available content.

Our goals:
1. For content smaller than the screen height, the empty state image should fill the empty space
2. For content higher than the screen height, only a small part of the empty state image should be visible
    a. ...but during _rubber banding_, it would be preferable to expose the rest of the empty state image
    
We have this figured out for Native, but are struggling on the Web because of the `FlatList` quirks.

My idea:

```tsx
import React from "react";
import {FlatListProps, ScrollView, ScrollViewProps} from "react-native";

type BasicFlatListProps<ItemT> =
  ScrollViewProps
  & Pick<FlatListProps<ItemT>, 'ListHeaderComponent' | 'data' | 'renderItem'>; // and more (?)...

type HybridFlatListMode = 'short' | 'long';

type HybridFlatListProps<ItemT> = Omit<BasicFlatListProps<ItemT>, 'ListHeaderComponent'> & {
  /**
   * If the item count is below this threshold, the list will be in the "short" mode and will render a
   * `ScrollableFlatList`. Otherwise, it will be in the "long" mode and will render a real flat list.
   */
  itemCountThreshold: number,

  renderHeader: (mode: HybridFlatListMode) => React.ReactElement,
};


function ScrollFlatList<TItem>(props: BasicFlatListProps<TItem>) {
  const {data, renderItem} = props;

  return (
    <ScrollView {...props}>
      {/* implementation... */}
    </ScrollView>
  )
}

function HybridFlatList<TItem>(props: HybridFlatListProps<TItem>) {
  const {data, renderHeader, itemCountThreshold} = props;

  const mode = data.length < itemCountThreshold ? 'short' : 'long';

  if (mode === 'short') {
    return <ScrollFlatList {...props} />;
  } else { // "real" mode
    return (
      <InvertedFlatList {...props}>
        {/* implementation... */}
      </InvertedFlatList>
    );
  }
}

/**
 * Cropped empty state image. I'm not sure what "cropped" means technically, but as the last resort, we could just use
 * another .jpg image.
 */
function CroppedEmptyStateImage(): React.JSX.Element {
  throw new Error("TODO: Implement");
}

/**
 * Position "absolute", etc.
 */
function ShiftedEmptyStateImage(): React.JSX.Element {
  throw new Error("TODO: Implement");
}

type EmptyStateFlatListProps<ItemT> = {
  readonly data: ReadonlyArray<ItemT>;
  readonly renderItem: FlatListProps<ItemT>['renderItem'];
}

// Web:
function EmptyStateFlatList<ItemT>({data, renderItem}: EmptyStateFlatListProps<ItemT>) {
  return (
    <HybridFlatList
      data={data}
      renderItem={renderItem}
      renderHeader={(mode) => {
        switch (mode) {
          case "short":
            return <ShiftedEmptyStateImage />;
          case "long":
            return <CroppedEmptyStateImage />;
        }
      }}
      itemCountThreshold={16} // arbitrary, let's pick it experimentally having Expensify one-line messages in mind
    />
  )
}

// Native:
function EmptyStateFlatList<ItemT>({data, renderItem}: EmptyStateFlatListProps<ItemT>) {
  return (
    <InvertedFlatList
      data={data}
      renderItem={renderItem}
      ListHeaderComponent={ShiftedEmptyStateImage}
    />
  )
}
```

This achieves all goals on nearly all platforms:
- On Native, we already have a working solution, so let's utilize it
- On the Web, let's implement a "fake" `FlatList` implementation for short lists, when we need the empty state image the most and when performance means the least
  - We lose the rubber banding effect, but only on the mWeb, as rubber banding is **disabled** in Expensify on desktop Web anyway
  
This is just an inspiration; let me know what you think about it and if you'd like to experiment on it. If you have any better ideas to achieve our goals with reasonable trade-offs, I'm open to hearing them!

**Comment by cubuspl42** [comment]
I'm not sure if this solution doesn't have any quirks; for example, I'm not sure if React will correctly maintain the identity of the item during the switch between the "hybrid modes". But if we keep the same item keys, I guess it _should_...

**Comment by cubuspl42** [comment]
It might be tricky to maintain the scrolling state; we might need to do some custom tricks to "move" it between the `ScrollableView` and `FlatList` 😕

It's kind of complicated, I won't lie, I wish we had something simpler. Eh, sometimes achieving a seemingly simple UI can be such a challange.

**Comment by shawnborton** [comment]
Does that mean this is ready to be re-tested again? Happy to spin up a test build if so.

**Comment by shawnborton** [comment]
Ah, I would consider that a blocker personally. Are we able to fix that?

**Comment by allroundexperts** [comment]
Hi @rayane-djouah!
Can you please resolve the conflicts here?

**Comment by allroundexperts** [comment]
@eh2077 Would be taking this over. I had to focus on other high priority issues.

**Comment by eh2077** [comment]
Spent some time to catch up previous discussions. 

@rayane-djouah Let me spend a bit more time to dig deeper and I'll feedback on your question about `Overflow part is not showing in the over-scroll area in mWeb Safari` later.

Can you please fix the lint error?

**Comment by eh2077** [comment]
> we have the "Overflow part is not showing in the over-scroll area in mWeb Safari" bug because of the behaviour of ScrollView (FlatList use ScrollView under the hood) as I explained https://github.com/Expensify/App/pull/38449#issuecomment-2041527898. Do you have any ideas on this?

The root cause of the inconsistency on mobile Safari is still not clear to me. Can you elaborate on it?
If we can't fix this in App, is it possible to fix it in the upstream?

**Comment by shawnborton** [comment]
Okay I think this is feeling pretty good and it's a major improvement, I think we can leave with the mWeb Safari bug where the rest of the BG image isn't exposed when you bounce scroll/stretch the view a bit. 

But that being said, there is still a visible gap between BG and the top of the page on mWeb, and I think we should at least address that: 
IMG_9815

**Comment by shawnborton** [comment]
Also cc @Expensify/design to test this one out too (when things are stable...)

**Comment by eh2077** [comment]
I'm having trouble to login on mobile device. Not sure if it's because of site issue. I'll continue testing when the site is full recovered.

**Comment by shawnborton** [comment]
Confirming that the Safari mWeb bug seems to be gone, nice! Let's get this into final review and hopefully merge soon.

**Comment by eh2077** [comment]
I'm re-testing it

**Comment by eh2077** [comment]
## Reviewer Checklist


- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/Expensify/App/assets/117511920/75286fc1-8fc2-4252-90d0-42845de90632


</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/Expensify/App/assets/117511920/c841d3bd-fcc0-4dd2-9a9f-73ced7239395


</details>

<details>
<summary>iOS: Native</summary>



https://github.com/Expensify/App/assets/117511920/ac5dcbd3-dbdc-41d5-b624-feafe20838af


</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/Expensify/App/assets/117511920/1dd5621c-db47-4791-ab5f-7f75882eca8d


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/Expensify/App/assets/117511920/9f28e214-3e76-478b-ab59-96ad02c89f29


</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/Expensify/App/assets/117511920/b2c9a160-eb17-4933-8bac-80b63d0af369


</details>

**Review by eh2077 — APPROVED** [review]
LGTM

**Comment by shawnborton** [comment]
Love it, cc @allroundexperts @yuwenmemon - who needs to do the final review here?

**Review by yuwenmemon — APPROVED** [review]

**Comment by yuwenmemon** [comment]
LGTM, just waiting on merge freeze to be lifted.

**Comment by pecanoro** [comment]
This PR most likely caused this issue https://github.com/Expensify/App/issues/42760 but I think we could fix it here: https://github.com/Expensify/App/pull/42629

**Comment by stitesExpensify** [comment]
I think it's possible that this PR also caused this regression https://github.com/Expensify/App/issues/42795. Specifically the part where the background UI is overlapping the report

**Comment by shawnborton** [comment]
Wow yeah this doesn't look great:
CleanShot 2024-05-30 at 20 24 13@2x

**Pull request merged**
